sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -785,5 +785,195 @@ private Request CreateGetRevisionsRequest(SettingSelector selector, string pageL
+
+                switch (response.Status)
+                {
+                    case 200:
+                        return response;
+                    case 404:
+                        throw await response.CreateRequestFailedExceptionAsync("Not found.").ConfigureAwait(false);
+                    case 412:
+                        throw await response.CreateRequestFailedExceptionAsync("The specified precondition failed.").ConfigureAwait(false);
+                    default:
+                        throw await response.CreateRequestFailedExceptionAsync().ConfigureAwait(false);
+                }
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Locks an existing <see cref="ConfigurationSetting"/> in the configuration store.
+        /// </summary>
+        /// <param name="key">The primary identifier of a configuration setting.</param>
+        /// <param name="label">The value used to group configuration settings.</param>
+        /// <param name="etag">The value of an etag indicates the state of a configuration setting within a configuration store.
+        /// If it is specified, the configuration setting is only locked if etag value matches etag value in the configuration store.
+        /// If no etag value is passed in, then the setting is always locked.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        public virtual Response Lock(string key, string label = default, ETag etag = default, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _pipeline.Diagnostics.CreateScope("Azure.Data.AppConfiguration.ConfigurationClient.Lock");
+            scope.AddAttribute("key", key);
+            scope.Start();
+
+            try
+            {
+                using Request request = CreateLockRequest(key, label, etag);
+                Response response = _pipeline.SendRequest(request, cancellationToken);
+
+                switch (response.Status)
+                {
+                    case 200:
+                        return response;
+                    case 404:
+                        throw response.CreateRequestFailedException("Not found.");
+                    case 412:
+                        throw response.CreateRequestFailedException("The specified precondition failed.");
+                    default:
+                        throw response.CreateRequestFailedException();
+                }
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private Request CreateLockRequest(string key, string label, ETag etag)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw new ArgumentNullException(nameof(key));
```

Comment:
I have some helpers coming out for this that will be in Azure.Core, but only if `key` is null should you throw `ArgumentNullException`. If empty, that should be `ArgumentException`. Could give the wrong idea, and this is consistent with the Framework/Core. You could also use Azure.Core's `Guard` class now and I'll be fixing it up later.

Reply:
@heaths, could you file an issue for me to update this once your tools become available?  We're handling this incorrectly elsewhere in the library and I'd like to fix it in one swoop.